vcpkg/concepts/features.md

```diff
@@ -0,0 +1,167 @@
+
+>![NOTE]
+> They are not intended to model "curation" or "suggestions".
+
+For example, consider a library `"extract-any"` that supports over 10 different archive formats, including several that are quite obscure. Because they are all optional, if none are selected the library is not functional: it cannot extract any files.
+
+Default features ensure that a user who simply adds `"extract-any"` to the list of dependencies in their `vcpkg.json` will get a baseline level of functionality, for example automatically selecting `.zip` and `.tar.gz` decompressors.
+
+### Example 2: Default features in action
+
+When a user adds `"extract-any"` to their `vcpkg.json` without specifying features, the default features (e.g., support for `.zip` and `.tar.gz` formats) are automatically included, ensuring basic functionality.
+
+```
+{
+  "name": "my-application",
+  "version": "0.15.2",
+  "dependencies": [
+    "extract-any"
+  ]
+}
+```
+
+If the user wants to explicitly disable the default features, they can do so by adding `"default-features": false` to the dependency:
+
+```json
+{
+  "name": "my-application",
+  "version": "0.15.2",
+  "dependencies": [
+    {
+      "name": "extract-any",
+      "default-features": false
+    }
+  ]
+}
+```
+
+Alternatively, if using vcpkg in [classic mode](../consume/classic-mode.md), you can disabled default features via the `core` feature. For example, `vcpkg install extract-any[core]` installs `extract-any` without any default features, as `[core]` explicitly excludes them.
+
+For more information, check out the [default features article](default-features.md).
+
+## Dependency resolution
+
+When using vcpkg, dependency resolution plays a crucial role, especially when dealing with features that have interdependencies. To illustrate, consider the following scenario involving an image manipulation library:
+
+
+```json
+{
+  "name": "my-image-lib",
+  "version": "0.1",
+  "features": {
+    "png": { "description": "Support PNG files", "dependencies": ["libpng"]},
+    "jpeg": { "description": "Support JPEG files", "dependencies": ["libjpeg-turbo"]},
+    "tiff": { "description": "Support TIFF files", "dependencies": ["libtiff"]},
+  }
+}
+```
+
+In scenarios where different libraries depend on various features of a common library, vcpkg ensures that all required features and dependencies are considered. For instance, if `library-a` requires the `png` feature and `library-b` requires the `jpeg` feature from `my-image-lib`, the dependency graph would look like this:
+
+```json
+{
+  "name": "library-a",
+  "version": "1",
+  "dependencies": [{"name": "my-image-lib", "features": ["png"]}]
+}
+```
+
+```json
+{
+  "name": "library-b",
+  "version": "1",
+  "dependencies": [{"name": "my-image-lib", "features": ["jpeg"]}]
+}
+```
```

Comment:
Just for completion, include a third manifest that declares dependencies in `library-a` and `library-b`.